server/src/main/java/org/elasticsearch/action/admin/indices/resolve/ResolveIndexAction.java

```diff
@@ -453,19 +451,27 @@ public TransportAction(
             ActionFilters actionFilters,
             IndexNameExpressionResolver indexNameExpressionResolver
         ) {
-            super(NAME, transportService, actionFilters, Request::new, EsExecutors.DIRECT_EXECUTOR_SERVICE);
-            this.clusterService = clusterService;
+            super(NAME, true, new Services() {
+                @Override
+                public TransportService transportService() {
+                    return transportService;
+                }
+
+                @Override
+                public ClusterService clusterService() {
+                    return clusterService;
+                }
+            }, actionFilters, Request::new, EsExecutors.DIRECT_EXECUTOR_SERVICE);
             this.remoteClusterService = transportService.getRemoteClusterService();
             this.indexNameExpressionResolver = indexNameExpressionResolver;
             this.ccsCheckCompatibility = SearchService.CCS_VERSION_CHECK_SETTING.get(clusterService.getSettings());
         }
 
-        @Override
-        protected void doExecute(Task task, Request request, final ActionListener<Response> listener) {
+        @ActionHandler
+        public void resolveIndices(final Request request, final ClusterState clusterState, final ActionListener<Response> listener) {
```

Comment:
I picked 1 simple action as an example.
There's nothing special about this action, other than it needs the cluster state, but doesn't need `ClusterService` for any other reason.